site_scons/buildutils.py

```diff
@@ -1643,3 +1648,147 @@ def make_relative_path_absolute(path_to_check: Union[str, Path]) -> str:
         pth = Path(Dir("#" + path_to_check).abspath)
 
     return pth.as_posix()
+
+
+def run_preprocessor(
+    conf: "SConfigure",
+    includes: "TextOrSequence",
+    text: "TextOrSequence",
+    defines: "TextOrSequence" = ()
+) -> Tuple[int, str]:
```

Comment:
This function could use a docstring. Especially to explain what the return value is.